### PR TITLE
Adding pic support for Kokkos

### PR DESCRIPTION
**Pull request by JulienLoiseau**
@junghans

### CODE REVIEW

**Comment by junghans** [comment]
@jjwilke

**Review by junghans — APPROVED** [review]

File: var/spack/repos/builtin/packages/kokkos/package.py [hunk 1]
```diff
@@ -208,6 +210,9 @@ def cmake_args(self):
         if isdiy:
             options.append("-DSpack_WORKAROUND=On")
 
+        if "+fpic" in spec: 
+            options.append("-DCMAKE_CXX_FLAGS=%s" % self.compiler.pic_flag)
```
Comment (junghans) [inline var/spack/repos/builtin/packages/kokkos/package.py]: Actually in CMake just use `-DCMAKE_POSITION_INDEPENDENT_CODE=ON`.

File: var/spack/repos/builtin/packages/kokkos/package.py [hunk 2]
```diff
@@ -172,6 +172,8 @@ class Kokkos(CMakePackage, CudaPackage):
     conflicts("+wrapper", when="~cuda")
 
     variant("std", default="11", values=["11", "14", "17", "20"], multi=False)
+    variant("fpic", default=False, description="Build position independent code")
```
Comment (junghans) [inline var/spack/repos/builtin/packages/kokkos/package.py]: ANd most other packages call this variant `pic` instead of `fpic`.

**Review by jjwilke — APPROVED** [review]

**Comment by jjwilke** [comment]
LGTM. Can't figure out why the CI is failing.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**